src/app/reporting/SynchronizedReportSchedulerImpl.h

```diff
@@ -42,17 +84,80 @@ class SynchronizedReportSchedulerImpl : public ReportSchedulerImpl, public Timer
 
     bool IsReportScheduled(ReadHandler * ReadHandler) override;
 
+    /** @brief Callback called when the report timer expires to schedule an engine run regardless of the state of the ReadHandlers,
+     *
+     * It loops through all handlers and sets their CanBeSynced flag to true if the current timstamp is greater than
```

Comment:
What is a "timstamp"?  Seriously, please run a spelling checker over all this text.